server/controller/Auth.js

```diff
@@ -1,94 +1,91 @@
-      }
 
-      // Check if the email is already registered
-      const existingUser = await User.findOne({ email });
-      if (existingUser) {
+      let user = await user.findOne({ email });
+  
+      if (user) {
         return res.status(400).json({ error: 'Email is already registered' });
       }
-
-      // Hash the password
+  
+      // Hashing the password
       const hashedPassword = await bcrypt.hash(password, 10);
-
-      // Create a new user
-      const newUser = new User({
+      
+      //SLY-13
+      user = new User({
         name,
         email,
         password: hashedPassword,
       });
 
-      // Save the user to the database
-      await newUser.save();
-
-      res.status(201).json({ message: 'User registered successfully' });
+      await user.save();
+       // SLY-14
+      // Generate JWT token
+      const token = jwt.sign({ userId: user._id }, process.env.JWT_SECRET, {
+        expiresIn: '48h',
+      });
+      //SLY-15
+      //SLY-24
+      res.status(201).json({ token });
     } catch (error) {
       console.error(error);
       res.status(500).json({ error: 'Error registering user' });
     }
-  }
-
-  async loginUser(req, res) {
+  };
+
+  //SLY-1
+  const loginUser = async (req, res) => {
+    // Validate request body
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+  
+    const { email, password } = req.body;
+  
     try {
-      // Validate input data
-      const { email, password } = req.body;
-
-      if (!email || !password) {
-        return res.status(400).json({ error: 'Please provide email and password' });
-      }
-
-      // Check if the email is in a proper format
-      const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
-      if (!emailRegex.test(email)) {
-        return res.status(400).json({ error: 'Please provide a valid email address' });
-      }
-
-      // Check if the user exists
       const user = await User.findOne({ email });
-
+  
       if (!user) {
         return res.status(401).json({ error: 'Invalid credentials' });
       }
-
+  
       // Compare the provided password with the stored hash
       const passwordMatch = await bcrypt.compare(password, user.password);
-
+  
       if (!passwordMatch) {
         return res.status(401).json({ error: 'Invalid credentials' });
       }
-
+      // SLY-11
       // Create a JSON Web Token (JWT)
-      const token = jwt.sign({ userId: user._id }, this.config.jwtSecret, {
-        expiresIn: '1h', // Token expires in 1 hour
+      const token = jwt.sign({ userId: user._id }, process.env.JWT_SECRET, {
+        expiresIn: '48h',
       });
-
+      //SLY-23
       res.status(200).json({ token });
     } catch (error) {
       console.error(error);
       res.status(500).json({ error: 'Error logging in' });
     }
-  }
-}
+  };
+
+  const logoutUser = (req, res) => {
```

Comment:
No API necessary for logout

Reply:
Can i remove the logout function then??

Reply:
yeah

Reply:
removed logout api

Reply:
I still see the logout function here, why?